test/compilable/test13336b.d

```diff
@@ -0,0 +1,22 @@
+// REQUIRED_ARGS: -o-
+// PERMUTE_ARGS:
+/* TEST_OUTPUT:
+---
+---
+*/
+int sx;
+double sy;
+
+ref f1(bool f)
+{
+    if (f)
+        return sx;
+    return sy;
```

Comment:
Why is this supposed to pass now? Other changes to tests look suspicious, too. Can you explain?

Reply:
Without my patch, if you have this code:

```d
int sx;
double sy;

double f1(bool f)
{
     if (f)
        return sx;
     return sy;
}
```

The code compiles. If we add `ref` to the return statement it doesn't compile anymore stating that "`cast(double)sx` is not an lvalue". Why wouldn't the cast be an lvalue? If we want to add other rules that prevent a specific class of casts then that should be discussed in a different context.

Reply:
We can discuss other changes in tests that look "suspicious" on a case by case basis.

Reply:
> Why wouldn't the cast be an lvalue?

You cannot return a reference to both an integer and a double at the same time.